lib/internal/streams/pipeline.js

```diff
@@ -189,8 +221,15 @@ function pipeline(...streams) {
       return;
     }
 
-    while (destroys.length) {
-      destroys.shift()(error);
+    while (
+      (hasTransform && destroys.length - 1) ||
+      (!hasTransform && destroys.length)
```

Comment:
I don’t have context here, so perhaps this is too nitpicky, but it might help readability if each of these two cases were set to variables describing what they mean.

Reply:
This would be great